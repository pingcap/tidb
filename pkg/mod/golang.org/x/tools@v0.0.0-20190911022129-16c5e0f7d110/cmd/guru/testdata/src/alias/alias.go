// Tests of Go 1.9 type aliases.
// See go.tools/guru/guru_test.go for explanation.
// See alias.golden for expected query results.

package alias // @describe describe-pkg "alias"

type I interface { // @implements implements-I "I"
	f()
}

type N int

func (N) f() {}

type M = N // @describe describe-def-M "M"
var m M    // @describe describe-ref-M "M"

type O N // @describe describe-O "O"

type P = struct{ N } // @describe describe-P "N"

type U = undefined // @describe describe-U "U"
type _ = undefined // @describe describe-undefined "undefined"
