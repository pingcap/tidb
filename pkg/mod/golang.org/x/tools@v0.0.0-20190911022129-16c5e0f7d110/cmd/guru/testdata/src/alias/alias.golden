-------- @describe describe-pkg --------
definition of package "alias"
	type  I interface{f()}
		method (I) f()
	type  M = N
		method (N) f()
	type  N int
		method (N) f()
	type  O int
	type  P = struct{N}
		method (struct{N}) f()
	type  U = invalid type
	var   m N

-------- @implements implements-I --------
interface type I
	is implemented by basic type N

-------- @describe describe-def-M --------
alias of type N (size 8, align 8)
defined as int
Methods:
	method (N) f()

-------- @describe describe-ref-M --------
alias of type N (size 8, align 8)
defined as int
Methods:
	method (N) f()

-------- @describe describe-O --------
definition of type O (size 8, align 8)
No methods.

-------- @describe describe-P --------
type struct{N} (size 8, align 8)
Methods:
	method (struct{N}) f()
Fields:
	N N

-------- @describe describe-U --------
alias of type invalid type

-------- @describe describe-undefined --------
identifier

